ast/ast.fbs

```diff
@@ -374,6 +381,11 @@ table ArrayExpression {
   elements:[WrappedExpression];
 }
 
+table VectorExpression {
+  base_node:BaseNode;
+  elements:[WrappedExpression];
+}
+
```

Comment:
Same comment as above. I'll give some explanation:

For arrays, we let users in Flux create an array using `[]`, as in `[1, 2, 3]` would be an `ArrayExpression` that has type `Array(int)`, or in our type syntax, the type would be expressed as `[int]`.

For for vectors, we don't want users to be able to create them. We want a special transformation of semantic graph function expressions to create them automatically for uses of `map` and `filter` in a user's script.

Most often, vectors will be created as part of a functions signature.  So instead of a function accepting a record type where `r` has scalar fields:
- E.g., `map(fn: (r) => ({a: r.a + 1}))`, where the type of `r` is `{a: int}`, in other words, "a record with a field called `a` whose type is `int`"
- The type of `r` will be changed to `{a: vector(int)}`, in other words, "a record with a field called a whose type is `vector` of `int`"

So there's no need to create an actual vector expression, we'll assume that the caller of the function being transformed will create the vector.

(There will be a need for a new expression called `expand` which will be able to create a `vector` out of a scalar value, but that is later work)